### DISCUSSION TITLE
Web3 not being recognized in the code

### DISCUSSION THREAD

**Discussion by ChristianG77** [smartcontractkit/full-blockchain-solidity-course-py#118] [Q&A] [answered]
Hi,

I installed the latest Build Tools and successfully completed pip install web3, however I'm receiving a message saying, 'Import "web3" could not be resolved.'

Also the web3 portion at the top of the code, 'from web3 import Web3', is not highlighting in green and is underlined.

Has anyone else had/solved this problem?

Thanks for your help.

**Comment by AlbertoSinigaglia** [comment, marked as answer]
if you run again `pip install web3` what is printed? which IDE are you using? are you using a virtual environment?

**Reply by ChristianG77** [reply]
Thanks Alberto, problem solved!

**Reply by PatrickAlphaC** [reply]
Boom! @AlbertoSinigaglia 💪👌🙏🙌

**Reply by PatrickAlphaC** [reply]
Thanks for posting @ChristianG77

**Comment by widmung89** [comment]
Hi, I also faced the same issue of "Import "web3" could not be resolved", when I entered 'pip install web3', I was also presented a long string of errors including "error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/". I followed some instructions on forums an downloaded and installed the Visual Studio Build Tools and tried doing "pip install web3'" again, it seemed to have installed but when I key in 'from web3 import Web3' into the code, I still the got message "Import "web3" could not be resolved".

**Reply by cromewar** [reply]
Hello there, in the bottom left corner on vscode you should see the python versión you are using, you must select the correct python interpreter on which you have installed web3.
Also if the interpreter is correct, you would like the restart vscode to let it charge the changes you installed.

Best regards, Cromewar.

**Reply by CSalle** [reply]
Hi @widmung89 , could you share how you solved this issue, please? Facing the same situation. Thanks a lot!

**Reply by cromewar** [reply]
@CSalle could you give us more information about what you have tried? Screenshots are always helpful.

**Reply by CSalle** [reply]
Hi @cromewar, thanks for your reply.

I figured out the solution to this issue. To whomever experienced it, you may try the following:

1. Install Visual Studio Build Tools [here](https://visualstudio.microsoft.com/visual-cpp-build-tools/)
2. Select the C++ Desktop development Workload for installation (should include "Visual C++ Build tools core features", "MSVC toolset C++ 2019 v142 (x86,x64)", "Visual C++ 2019 Redistributable Update" and "Windows 10 SDK (10.0.17763.0) for Desktop C++")
3. Finish installation and reboot laptop
4. Open Visual Studio Code and retry pip install web3

That should do it :)

Thanks a lot!

**Reply by cromewar** [reply]
You are awesome @CSalle I'll take this in mind for further problems.

**Reply by david-randoll** [reply]
> [CSalle ](/CSalle)

Thank you! This worked for me. Took about 6 GB of storage

**Reply by eiyen** [reply]
@CSalle 
It's also worked for me! Thanks a lot, i've been stuck here for over 3 days. I chose a little more choices, it's about 12 GB of storage for me.

**Reply by formulaRoot** [reply]
> 

This solved my issue. I had a couple of versions of python and the IDE was using an older version I just changed it and boom, all errors disappeared !Thanks @cromewar
